src/create_atoms.cpp

```diff
@@ -845,7 +879,11 @@ void CreateAtoms::add_random()
       if (mode == ATOM) {
         atom->avec->create_atom(ntype, xone);
       } else {
-        add_molecule(xone);
+
+        // atomic coordinates calculated above for overlap check
+
+        if (!overlapflag) get_xmol(xone);
+        add_molecule();
```

Comment:
Also, does this logic in this code section mean that the proc which owns the center pt of the molecule is adding all atoms in the molecule to it's list of atoms?  It seems like for a large molecule (which extended over multiple proc subdomains), that could lead to lost atoms when comm->exchange() is eventually called.  If that's the case, maybe it would be better for each proc to only add the atoms from the molecule which are in its subdomain?

Reply:
good question. You are suggesting it is a bug prior to this PR, right? If so, I would be surprised that no one has tried to add a large molecule (I know I have, but maybe not in parallel).

Reply:
yes, this is a prior bug
the same issue exists for adding molecules on a lattice via get_xmol() and add_molecule() in loop_lattice()
I don't see an easy way to solve the problem there since multiple procs are inserting different molecules simultaneously

However for random and single, the insertion is one molecule at a time.

So I think a good solution is to add a 2nd add_molecule(double **sublo, double **subhi)
method and have add_single() and add_random() call that one instead when mode == MOLECULE.
In the new method, loop over the atoms in the molecule and put the logic for testing against sublo/hi
inside the loop, so that each proc adds only the atoms it owns.  So the caller does not do the test for xone
against sublo/subhi when mode == MOLECULE.

And we should clarify on the doc page that if the lattice method is used to insert mols 2x larger (in any dim) than proc subdomain sizes, lost atoms could result.  But that should not be a problem for single and random mode.